docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
 
 Next, you will want to go through the [Acceptable Formulae](Acceptable-Formulae.md) or [Acceptable Casks](Acceptable-Casks.md) documentation to determine if the software is an appropriate addition to Homebrew. If you are creating a formula for an alternative version of software already in Homebrew (e.g. a major/minor version that differs significantly from the existing version), be sure to read the [Versions](Versions.md) documentation to understand versioned formulae requirements.
 
-If everything checks out, you're ready to get started on a new formula!
+If everything checks out, you're ready to get started on a new formula / cask!
+
+**Note:** For updating the version of an existing formula or cask, refer to the [Updating Software in Homebrew](Updating-Software-in-Homebrew.md) guidelines, as well as 'submitting a new version of a [formula](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-formula) / [cask](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-cask)' in 'How To Open A Homebrew Pull Request'.
 
 ## Formulae
 
+**Note:** Before taking the time to craft a new formula:
+
+* make sure it can be accepted by checking [Acceptable Formulae](Acceptable-Formulae.md).
+* check that the formula was not [already refused](https://github.com/Homebrew/homebrew-core/search?q=is%3Aclosed&type=Issues).
+* if you are just updating the version of an existing formula, see the streamlined method in ['submit a new version of an existing formula'](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-formula).
+
```

Comment:
The first and last bullet here repeat links stated in the section before so feel a little redundant.

Reply:
Yeah, I thought about that. Originally the content was at the top of the page only. Which in one sense is good, as that's the first place people will look. But then sometimes I know that I will skip straight to the cask/formula section and not look at the rest of the page (doubly so because so many of the Homebrew docs pages are super long/convoluted, so I just try and skim to the bits that seem relevant at the time)

I don't have particularly strong feelings one way or the other here. Whichever is most likely to be seen/followed by the larger number of people was my intent.

Reply:
@MikeMcQuaid Do you have a preference based on my comments above? Should we keep this at the top of the file, or at the top of each cask/formula section?

Reply:
> Should we keep this once at the top of the file, or at the top of each cask/formula section?

I don't have strong feelings either way beyond: let's not do it in both places.

Reply:
Ok :) Barring anyone expressing strong feelings otherwise, I'll remove it from the top area, and keep the ones in the Formula/Cask sections.